src/dataframe/dataframe.jl

```diff
@@ -98,6 +98,10 @@ use the functionality provided by `select`/`transform`/`combine` functions, use
 functions, or provide type assertions to the variables that hold columns
 extracted from a `DataFrame`.
 
+Metadata: this function preserves all table and column level metadata.
+As a special case if `GroupedDataFrame` is passed to `DataFrame` then
+only `:note` style metadata from parent of the `GroupedDataFrame` is preserved.
```

Comment:
The problem is that `GroupedDataFrame` can be subsetted, i.e. you can drop groups from it.

For this reason, at least for the time being, I thought to decide that `GroupedDataFrame` does not have metadata itself and also that `DataFrame` constructor does not propagate metadata in this case.

However, it is debatable. Initially I had design where `GroupedDataFrame` had `:note`-style only metadata and `DataFrame` constructor propagated it. So let us discuss which approach is better.

Reply:
Allowing to set/get metadata from `GroupedDataFrame` is indeed not super important, at least right now. But given that `select(groupby(df, key), ...)` can propagate `:note` metadata, it would seem consistent for `DataFrame(groupby(df, key))` to do the same, right?

Reply:
But then `DataFrame(groupby(df, key))` should keep `:note`-style metadata only. Right?

Reply:
Yes I guess.

Reply:
So this is what we do now. The reason is that this operation is in the end:
```
    if keepkeys
        return parent(gd)[idx, :]
    else
        return parent(gd)[idx, Not(gd.cols)]
    end
```
so it is just `getindex`, which keeps `:note`-style metadata, but drops all other metadata.

BTW: note that this constructor even allows dropping grouping columns from the resulting data frame (I do not remember why it was requested, but someone wanted it 😄).